services/viva-ms/lambdas/syncWorkflow.js

```diff
@@ -12,39 +11,65 @@ import * as dynamoDb from '../../../libs/dynamoDb';
 
 const SSMParams = params.read(config.vada.envsKeyName);
 
-const dynamoDbConverter = AWS.DynamoDB.Converter;
-
 const CASE_WORKFLOW_PATH = 'details.workflow';
 
 export async function main(event) {
-  if (event.detail.dynamodb.NewImage === undefined) {
-    return null;
-  }
+  const personalNumber = event.detail.user.personalNumber;
+  const PK = `USER#${personalNumber}`;
 
-  const { PK, SK, details } = dynamoDbConverter.unmarshall(event.detail.dynamodb.NewImage);
-  const personalNumber = PK.substring(5);
-  const { workflowId, workflow: caseWorkflow } = details;
+  const allUserCases = await getAllUserCases(PK, PK);
 
-  if (!workflowId) {
-    return false;
-  }
+  await syncCaseWorkflows(allUserCases, personalNumber);
 
-  const [vadaMyPagesError, vadaMyPagesResponse] = await to(
-    sendVadaMyPagesRequest(personalNumber, workflowId)
-  );
-  if (vadaMyPagesError) {
-    return console.error('(Viva-ms) syncWorkflow VADA request error', vadaMyPagesError);
-  }
+  return true;
+}
+
+async function getAllUserCases(PK, SK) {
+  const TableName = config.cases.tableName;
 
-  const vivaWorkflow = vadaMyPagesResponse.attributes;
+  const params = {
+    TableName,
+    KeyConditionExpression: 'PK = :pk AND begins_with(SK, :sk)',
+    ExpressionAttributeValues: {
+      ':pk': PK,
+      ':sk': SK,
+    },
+  };
```

Comment:
If we want all cases on a user, isn't it enough to just look at the PK? All cases for a given user should have the same PK, so we don't need to care at all about the SK here, and can simplify the params to just ask for 'PK = :pk' and nothing more.

Reply:
Good point! 👍
Changed query to only contain PK.